### ISSUE TITLE
Logic Analyzer: Trigger activates on the second event

### ISSUE THREAD

**Issue by drbautista** [analogdevicesinc/scopy#219]
OS: Windows 7 64bit
 Scopy Version:  454babc
 Firmware Version: v0.14

*Open Digital IO instrument [This will serve as the trigger]
1. Set DIO0 as output and put it on Logic 0.
2. Run the Digital IO instrument

![digital io](https://user-images.githubusercontent.com/20266602/29015062-ec156c5e-7b7e-11e7-92ce-697e48c9db1d.png)

*Open the Pattern Generator instrument.
2. Set DIO1 to DIO8 to clock pattern and assign any frequency [any pattern will also work].
       -For this case use frequencies between 10Hz to 500Hz
3. Run the pattern generator

![pattern generator](https://user-images.githubusercontent.com/20266602/29014985-82b5d4a6-7b7e-11e7-8808-752275aa9b60.png)

*Open Logic Analyzer instrument
3. Set time base to 5ms and position to 0ms
4. Set trigger to normal and Trigger Logic to OR
5. Set the trigger for DIO0 to rising edge
6. Click Single capture and see if the logic analyzer is waiting for a trigger.

![logic analyzer](https://user-images.githubusercontent.com/20266602/29015087-16ccb81c-7b7f-11e7-8385-e87a10c35f0e.png)

7. Go back to the digital IO instrument and change the output of DIO0 from 0 to 1.
8. Go back to the Logic Analyzer instrument and notice that it didn't trigger the capture.
9. Try to change the DIO0's output from 1 to 0 then 0 to 1 again.
10. That's the time it will trigger a capture.

*Same behavior happens on falling edge and any edge trigger option on all the DIO channels.
*Same behavior also happens using the external trigger
*I also tried using different source for triggering, and the behavior is still the same for the logic analyzer

**Comment by drbautista** [comment]
I have an additional observation regarding this bug.

The logic analyzer actually captures the signal on the first event but it will only show this captured signal on the second event of the trigger.

here's how I figured it out.

1. Set DIO9 as output and set to 0 on Digital IO instrument [this will be the trigger]
2. In the pattern generator instrument, group DIO0 to DIO7 and set as number pattern for this case I set a numerical value of 50. DIO8 will be a clock pattern at 100Hz

![pattern generator 1](https://user-images.githubusercontent.com/20266602/29059254-47453974-7c47-11e7-85c4-dd0b93f712a1.png)

3. In the logic analyzer instrument, group DIO0 to DIO8, and set as parallel. Set DIO0:DIO0 . . . DIO7:DIO7 and DIO8 as CLK. Set rising edge trigger on DIO9.
4. Trigger the rising edge of DIO9 in the Digital IO instrument, and see that the logic analyzer is not showing any captured data.
5. Change in the pattern generator the numerical value, this case is 100. 

![pattern generator 2](https://user-images.githubusercontent.com/20266602/29059268-546ca574-7c47-11e7-89d6-710f09505f86.png)

6. Trigger DIO9 again.

![logic analyzer result](https://user-images.githubusercontent.com/20266602/29059286-689c51ca-7c47-11e7-8f23-5802973456e9.png)

It shows the hex value of the Decimal 50 which is 32.
